wpilibj/src/test/java/edu/wpi/first/wpilibj/RobotDriveTest.java

```diff
@@ -0,0 +1,180 @@
+      }
+    }
+  }
+
+  @Test
+  public void testTankDrive() {
+    m_differentialDrive.setDeadband(0.0);
+    for (double leftJoystick : m_testJoystickValues) {
+      for (double rightJoystick : m_testJoystickValues) {
+        m_robotDrive.tankDrive(leftJoystick, rightJoystick, false);
+        m_differentialDrive.tankDrive(leftJoystick, rightJoystick, false);
+        assertEquals("Left Motor didn't match. Left Joystick: " + leftJoystick
+            + " Right Joystick: " + rightJoystick + " ", m_rdFrontLeft.get(), m_frontLeft.get(),
+            0.01);
+        assertEquals("Right Motor didn't match. Left Joystick: " + leftJoystick
+            + " Right Joystick: " + rightJoystick + " ", m_rdFrontRight.get(), m_frontRight.get(),
+            0.01);
+      }
+    }
+  }
+
+  @Test
+  public void testArcadeDriveSquared() {
+    m_differentialDrive.setDeadband(0.0);
+    for (double moveJoystick : m_testJoystickValues) {
+      for (double rotateJoystick : m_testJoystickValues) {
+        m_robotDrive.arcadeDrive(moveJoystick, rotateJoystick);
+        m_differentialDrive.arcadeDrive(moveJoystick, -rotateJoystick);
+        assertEquals("Left Motor squared didn't match. Move Joystick: " + moveJoystick
+            + " Rotate Joystick: " + rotateJoystick + " ", m_rdFrontLeft.get(), m_frontLeft.get(),
+            0.01);
+        assertEquals("Right Motor squared didn't match. Move Joystick: " + moveJoystick
+            + " Rotate Joystick: " + rotateJoystick + " ", m_rdFrontRight.get(),
+            m_frontRight.get(), 0.01);
+      }
+    }
+  }
+
+  @Test
+  public void testArcadeDrive() {
+    m_differentialDrive.setDeadband(0.0);
+    for (double moveJoystick : m_testJoystickValues) {
+      for (double rotateJoystick : m_testJoystickValues) {
+        m_robotDrive.arcadeDrive(moveJoystick, rotateJoystick, false);
+        m_differentialDrive.arcadeDrive(moveJoystick, -rotateJoystick, false);
+        assertEquals("Left Motor didn't match. Move Joystick: " + moveJoystick
+            + " Rotate Joystick: " + rotateJoystick + " ", m_rdFrontLeft.get(), m_frontLeft.get(),
+            0.01);
+        assertEquals("Right Motor didn't match. Move Joystick: " + moveJoystick
+            + " Rotate Joystick: " + rotateJoystick + " ", m_rdFrontRight.get(),
+            m_frontRight.get(), 0.01);
+      }
+    }
+  }
+
+  @Ignore("Needs more analysis")
+  @Test
+  public void testMecanumPolar() {
+    m_mecanumDrive.setDeadband(0.0);
+    for (double magnitudeJoystick : m_testJoystickValues) {
+      for (double directionJoystick : m_testJoystickValues) {
+        for (double rotationJoystick : m_testJoystickValues) {
+          m_robotDrive.mecanumDrive_Polar(magnitudeJoystick, directionJoystick, rotationJoystick);
+          m_mecanumDrive.drivePolar(magnitudeJoystick, directionJoystick,
+                                    rotationJoystick);
+          System.err.println(magnitudeJoystick + ", " + directionJoystick + ", " + rotationJoystick
```

Comment:
Probably not a good idea to print (to system.err no less) every iteration

Reply:
Why? This is a test, not actual robot code. By printing the output, it gives us more context when one of the asserts fails.

Reply:
Yes but this will print out even if the assert passes. Why not add it to an assert as the error message, so it only gets printed out when it fails?

Reply:
The assert only prints information about a single motor. Theoretically, it would be possible to print info for all motors, but that adds a lot to each of the 4 asserts. System.err is redirected into the test report. Note that it currently doesn't print anything, as the test is ignored.